### ISSUE TITLE
High VRAM consumption

### ISSUE THREAD

**Issue by lhallee** [TorchJD/torchjd#137]
Hello,

Very interesting work!

I have a setup that uses 5 losses for a certain objective. I am passing the losses to tensors and parameters to inputs in the backward function. Is this correct? I am getting at least 3x more VRAM being used with this vs. a typical backward pass.

```
if self.use_jd: # 3x more VRAM
    backward(
        tensors=losses,
        inputs=list(self.model.parameters()),
        A=self.aggregator
    )
else:
    total_loss = sum(losses)
    total_loss.backward()
```

**Comment by ValerianRey** [comment]
Hi! Yes, that's normal. In its current state, torchjd is not optimized for memory usage. Memory should scale roughly linearly with the number of losses. We'll try to improve that in the long term.

At the moment, there's not much that I can do to help you. Maybe if you have some losses that have very low conflict (their gradients are generally aligned) you can sum them, while keeping the strongly conflicting (negative cosine similarity between their gradients) losses separated. That could make you go down to two terms to differentiate separately, limiting the memory overhead.

**Comment by lhallee** [comment]
Got it. That's a good idea! Thanks
